hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/rsql/RSQLUtility.java

```diff
@@ -161,6 +167,109 @@ private static Node parseRsql(final String rsql) {
         }
     }
 
+
+    private static final class ValidationRSQLVisitor<A extends Enum<A> & FieldNameProvider> implements RSQLVisitor<Boolean, String>{
+
+        private Class<A> enumType;
+
+        public ValidationRSQLVisitor(final Class<A> enumType){
+            this.enumType = enumType;
+        }
+
+        @Override
+        public Boolean visit(AndNode node, String param) {
+            for(Node child: node.getChildren()){
+                if(!child.accept(this, param))
+                    return false;
+            }
+            return true;
+        }
+
+        @Override
+        public Boolean visit(OrNode node, String param) {
+            for(Node child: node.getChildren()){
+                if(child.accept(this, param))
```

Comment:
I'm not sure if the boolean algebra semantic is valid here... I guess AndNode and OrNode should behave same here, because you are checking the validity of the expression, not the condition evaluation. Could you add some tests to verify it?

Reply:
Good catch, you are absolutely correct here! Fixed.